src/ImageSharp/Formats/Tiff/Compression/Decompressors/DeflateTiffCompression.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.IO.Compression;
+
+using SixLabors.ImageSharp.Compression.Zlib;
+using SixLabors.ImageSharp.Formats.Experimental.Tiff.Constants;
+using SixLabors.ImageSharp.IO;
+using SixLabors.ImageSharp.Memory;
+
+namespace SixLabors.ImageSharp.Formats.Experimental.Tiff.Compression.Decompressors
+{
+    /// <summary>
+    /// Class to handle cases where TIFF image data is compressed using Deflate compression.
+    /// </summary>
+    /// <remarks>
+    /// Note that the 'OldDeflate' compression type is identical to the 'Deflate' compression type.
+    /// </remarks>
+    internal class DeflateTiffCompression : TiffBaseDecompresor
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DeflateTiffCompression" /> class.
+        /// </summary>
+        /// <param name="memoryAllocator">The memoryAllocator to use for buffer allocations.</param>
+        /// <param name="width">The image width.</param>
+        /// <param name="bitsPerPixel">The bits used per pixel.</param>
+        /// <param name="predictor">The tiff predictor used.</param>
+        public DeflateTiffCompression(MemoryAllocator memoryAllocator, int width, int bitsPerPixel, TiffPredictor predictor)
+            : base(memoryAllocator, width, bitsPerPixel, predictor)
+        {
+        }
+
+        /// <inheritdoc/>
+        protected override void Decompress(BufferedReadStream stream, int byteCount, Span<byte> buffer)
+        {
+            long pos = stream.Position;
+            using (var deframeStream = new ZlibInflateStream(
+                stream,
+                () =>
```

Comment:
@brianpopow @IldarKhayrutdinov What do you think of this pattern used by the inflate stream? I find it really odd, it was contributed by someone else a few years back.

Reply:
I'd love to get this finished tbh and either reference it or add the coed directly.
https://github.com/SixLabors/ZlibStream

Reply:
> @brianpopow @IldarKhayrutdinov What do you think of this pattern used by the inflate stream? I find it really odd, it was contributed by someone else a few years back.

yeah I also find it odd, somehow I tried to figure it out, but I decided don't change anything

> I'd love to get this finished tbh and either reference it or add the code directly.
> https://github.com/SixLabors/ZlibStream

great idea, it's we can compare performance and add `SixLabors/ZlibStream` as submodule